### PR TITLE
fix(ESSNTL-4399, ESSNTL-4405, ESSNTL-4402): Fix updated_start & updated_end edge-cases

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [ESSNTL-4399](https://issues.redhat.com/browse/ESSNTL-4399), [ESSNTL-4405](https://issues.redhat.com/browse/ESSNTL-4405), and [ESSNTL-4402](https://issues.redhat.com/browse/ESSNTL-4402):

- It allows for the start and end dates to be equal
- It fixes the bulk-delete validation so it also considers updated_start and updated_end
- It adds exception handling to the `GET /tags` to convey a useful message in case of user error

This follows up on the original updated_start and updated_end PR (#1272).

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [x] Tests: validate exceptions and failure scenarios
- [x] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [x] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: api/tag.py [hunk 1]
```diff
@@ -86,20 +88,24 @@ def get_tags(
...
+            fqdn,
+            display_name,
+            hostname_or_id,
+            insights_id,
+            provider_id,
+            provider_type,
+            updated_start,
+            updated_end,
+            tags,
+            None,
```
Comment (thearifismail) [inline api/tag.py:102]:
```suggestion
            None,    # Staleness to not use the default values
```

Comment (thearifismail) [reply api/tag.py:102]: For clarity, suggesting this comment.  I wondered what parameter had None value

**Review by thearifismail — COMMENTED** [review]
A minor comment

File: api/tag.py (hunk 1 above)
Comment (kruai) [reply api/tag.py:102]: Just to be clear, all I changed was that I indented the existing code here. But if you think the comment will help then sure

**Comment by fstavela** [comment]
@kruai I've tested this and it works fine. If you run pre-commit and unit tests locally and they pass, I think we can merge this.

**Review by fstavela — APPROVED** [review]

**Pull request merged**